### PR TITLE
Add supported_cultures liquid filter

### LINKED ISSUE #16202: Hard to implement a ContentCulturePicker in Liquid
@sebastienros 

> ContentCulturePicker.cshtml should resolve the cultures like before, and then invoke a new ContentCulturePickerContainer with this model. The driver can then use the ContentCulturePickerContainer instead, and rendering ContentCulturePicker will work like before.

no, it does not in 2.0.0-preview-18222: 
before, it was possible to override the shape with liquid and get all the required data as part of the Model.

I guess the problem is, that `ContentCulturePickerContainer` is not invoked anymore, so you directly need to override `ContentCulturePicker`

As the `ContentCulturePicker` does not work with a Model, but creates variables with the required data inside the template itself, there is no way to access this data inside a liquid template, which overrides the template.

Am I missing a way to call this in liquid to get the equivalent for `Model.SupportedCultures`?
`var supportedCultures = (await LocalizationService.GetSupportedCulturesAsync()).Select(c => CultureInfo.GetCultureInfo(c));`

_Originally posted by @bashuss in https://github.com/OrchardCMS/OrchardCore/issues/15148#issuecomment-2134979300_

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #16202

#### Usage Example

```
{% assign cultures = Culture | supported_cultures %}

<ul>
{% for culture in cultures %}
    <li>{{ culture.Name }}</li>
{% endfor %}
</ul>
```

### CODE REVIEW

File: src/docs/releases/2.0.0.md [hunk 1]
```diff
@@ -535,3 +535,17 @@ Previously, achieving the same result required more code:
 Before this release, the `MarkdownField` used a single-line input field by default (named the Standard editor) and offered two different editors: Multi-line with a simple `textarea` and WYSIWYG with a rich editor. Now, by default, it uses a `textarea` as the Standard editor, and the separate Multi-line option has been removed.
 
 You have nothing to do, fields configured with the Standard or Multi-line editors previously will both continue to work. Note though, that their editors will now be a `textarea`.
+
+### Liquid
+
+New filter named `supported_cultures` was added to allow you to get a list of supported cultures. Here is an example of how to print the name of supported cultures using a list
```
Comment (Piedone) [inline src/docs/releases/2.0.0.md:541]:
```suggestion
The `Culture` context variable got the new properties for a fuller Liquid culture support: `DisplayName`, `NativeName`, and `TwoLetterISOLanguageName`. These are the same as their .NET counterparts.

A new filter named `supported_cultures` was added to allow you to get a list of supported cultures. Here is an example of how to print the names of supported cultures using a list:
```

File: src/docs/reference/modules/Liquid/README.md [hunk 2]
```diff
@@ -504,8 +504,19 @@ The following properties are available on the `Culture` object.
...
+| `NativeName` | `English (United States)` | The native name of the current culture. |
+| `TwoLetterISOLanguageName` | `en` | The two-letter ISO language name of the current culture. |
+
+##### supported_cultures filter
+
+Returns the currently supported cultures.
+
+```liquid
+{{ Culture | supported_cultures }}
+```
```
Comment (Piedone) [inline src/docs/reference/modules/Liquid/README.md:519]: Note that the suggestion apparently can't include too many ` characters, so the trailing fence is missing but is there in the code if you try to edit this comment:

```suggestion
Returns the currently supported cultures. Here is an example of how to print the names of supported cultures using a list:

```liquid
{% assign cultures = Culture | supported_cultures %}
<ul>
{% for culture in cultures %}
    <li>{{ culture.Name }}</li>
{% endfor %}
</ul>
```
```

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Startup.cs
[the remark this answers is not in the document]
Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Startup.cs:1]: This is actually what we need to fix the linked issue. Why would we put it into a separate PR?

Or this PR can remain as it is, but then with the issue link removed.

But I don't see why we'd postpone this when the solution is here, only a copy-paste is needed.

[the remark this answers is not in the document]
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Startup.cs:1]: Both actually provide a fix for the issue "Hard to implement a ContentCulturePicker in Liquid". giving the user the helpers in this PR allows them to implement the template as they need.  Undoing the other changes also fixes the issue by providing all the variables in the injected model. 

I don't have the time to test out the proposed changes which is also why I want to leave them out of this PR.

[the remark this answers is not in the document]
Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Startup.cs:1]: Technically yes, but implementing the template is still harder than it was, and than it needs to be. I just don't see why we'd need the bureaucracy of another PR when it's the same topic, same issue.

But I opened a PR just for this, then: https://github.com/OrchardCMS/OrchardCore/pull/16230

**Review by Piedone — APPROVED** [review]

**Pull request merged**